lib/routes/recovery-key.js

```diff
@@ -171,18 +198,41 @@ module.exports = (log, db, Password, verifierVersion, customs) => {
       handler(request) {
         log.begin('recoveryKeyDelete', request)
 
-        return Promise.resolve()
-          .then(() => {
-            const sessionToken = request.auth.credentials
+        const sessionToken = request.auth.credentials
 
-            if (sessionToken.tokenVerificationId) {
-              throw errors.unverifiedSession()
-            }
+        function deleteRecoveryKey() {
+          if (sessionToken.tokenVerificationId) {
+            throw errors.unverifiedSession()
+          }
 
-            return db.deleteRecoveryKey(sessionToken.uid)
-              .then(() => {
-                return {}
-              })
+          return db.deleteRecoveryKey(sessionToken.uid)
+        }
+
+        function sendNotificationEmail() {
+          const geoData = request.app.geo
+          const ip = request.app.clientAddress
+          const emailOptions = {
+            acceptLanguage: request.app.acceptLanguage,
+            ip: ip,
+            location: geoData.location,
+            timeZone: geoData.timeZone,
+            uaBrowser: request.app.ua.browser,
+            uaBrowserVersion: request.app.ua.browserVersion,
+            uaOS: request.app.ua.os,
+            uaOSVersion: request.app.ua.osVersion,
+            uaDeviceType: request.app.ua.deviceType,
+            uid: sessionToken.uid
+          }
+
+          return db.account(sessionToken.uid)
+            .then((account) => mailer.sendPostRemoveAccountRecoveryNotification(account.emails, account, emailOptions))
+        }
+
+        return Promise.resolve()
+          .then(deleteRecoveryKey)
+          .then(sendNotificationEmail)
+          .then(() => {
+            return {}
```

Comment:
Not sure whether we're bothered about consistency here, but in the other route handlers the promise chain comes before the function definitions it uses. I find that way a bit easier to read fwiw, top-down rather than bottom-up.

Reply:
Will update!